### PR TITLE
refactoring and gerrit output for the parse subcommand

### LINKED ISSUE #2745: add gerrit output to the parse subcommand
Today 
diff command supports gerrit output, please add similar output fromat for the parse subcommand too to be able to convert all reports to gerrit.

### PR DESCRIPTION
**Pull request by gyorb**
resolves #2745

### CODE REVIEW

**Comment by gyorb** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- web/client/codechecker_client/cmd_line_client.py  1
- codechecker_common/tests/test_gerrit_converter.py  1
- analyzer/codechecker_analyzer/cmd/parse.py  2
- codechecker_common/output/gerrit.py  1
         

Complexity decreasing per file
==============================
+ codechecker_common/output/codeclimate.py  -1
         

Clones added
============
- codechecker_common/tests/test_gerrit_converter.py  13
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/CodeChecker/codechecker/pullRequest?prid=6423989&bid=20810377)

**Review by csordasmarton — CHANGES_REQUESTED** [review]

File: analyzer/codechecker_analyzer/cmd/parse.py [hunk 1]
```diff
@@ -375,7 +378,15 @@ def get_argparser_ctor_args():
         'epilog': """
 environment variables:
   CC_SEVERITY_MAP_FILE   Path of the checker-severity mapping config file.
+  CC_REPO_DIR         Root directory of the sources, i.e. the directory where
+                      the repository was cloned. Use it when generating gerrit
+                      output.
+  CC_REPORT_URL       URL where the report can be found. Use it when generating
+                      gerrit output.
+  CC_CHANGED_FILES    Path of changed files json from Gerrit. Use it when
+                      generating gerrit output.
```
Comment (bruntib) [inline analyzer/codechecker_analyzer/cmd/parse.py:387]: Where are these environment variables used in the code? I can't find it with grep under "analyzer" directory.

Comment (gyorb) [reply analyzer/codechecker_analyzer/cmd/parse.py:387]: They are used by the gerrit output converter here: codechecker_common/output/gerrit.py which is included by the analyzer part and the web part too.

**Review by csordasmarton — APPROVED** [review]
LGTM!

**Pull request merged**